ext/php5/serializer.c

```diff
@@ -39,10 +45,16 @@ static int write_hash_table(mpack_writer_t *writer, HashTable *ht TSRMLS_DC) {
                 mpack_start_array(writer, zend_hash_num_elements(ht));
             }
         }
+        bool string_as_uint64 = false;
         if (key_type == HASH_KEY_IS_STRING) {
             mpack_write_cstr(writer, string_key);
+            // If the key is trace_id, span_id or parent_id then strings have to be converted to uint64 when packed.
+            if (0 == strcmp(trace_id_keyword, string_key) || 0 == strcmp(span_id_keyword, string_key) ||
+                0 == strcmp(parent_id_keyword, string_key)) {
+                string_as_uint64 = true;
```

Comment:
I like the idea a lot. I am accepting this suggestion, but since I think it conflict with the BL below, I might have to change  bit later. Overall great idea at not changing the `msgpack_write_zval` signature!

Reply:
I removed the `string_as_uint64 ` from the method signature (thanks for thinking about this). I made a small change to your proposal, keeping the key serialization and the value serialization separate, thought. It looked more readable to me but I can be wrong. If you do not like it, please let me know as I have not a super strong opinion on this.